pkg/chunk/elastic/es_table_client.go

```diff
@@ -0,0 +1,77 @@
+package elastic
+
+import (
+	"context"
+	"github.com/olivere/elastic"
+
+	"github.com/pkg/errors"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+//var ctx = context.Background()
+
+type tableClient struct {
+	cfg    Config
+	client *elastic.Client
+}
+
+// NewTableClient returns a new TableClient.
+func NewTableClient(ctx context.Context, cfg Config) (chunk.TableClient, error) {
+	client, err := newES(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &tableClient{
+		cfg:    cfg,
+		client: client,
+	}, nil
+}
+
+// ListTables means list index in ElasticSearch
+func (c *tableClient) ListTables(ctx context.Context) ([]string, error) {
+	response, err := c.client.IndexNames()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return response, nil
+}
+
+func (c *tableClient) CreateTable(ctx context.Context, desc chunk.TableDesc) error {
+	//// Use the IndexExists service to check if a specified index exists.
```

Comment:
If you intend this to be a no-op call and use Elastics auto index management with templates, please clear out this commented code and replace with a comment note on how tables are created.